fix(stat): Collection of fixes for stat(1)

Alonely0 opened the pull request:
This PR fixes #9071, #9072, as well as a few bugs found along the way.

sylvestre commented:
a bunch of jobs are still failing

sylvestre commented:
See https://app.codecov.io/gh/uutils/coreutils/pull/9078?dropdown=coverage&src=pr&el=h1 for the tests improvements

sylvestre commented on src/uu/stat/src/stat.rs:
```diff
@@ -299,6 +301,16 @@ fn group_num(s: &str) -> Cow<'_, str> {
     res.into()
 }
 
+/// Keeps major part of an integer
```
The `major()` and `minor()` functions use incorrect bit-masking (`(n >> 8) & 0xFF` and `n & 0xFF`) that only handles old-style 8-bit device numbers. Modern Linux uses complex encoding with 12-bit major (up to 20-bit) and 8-bit minor (up to 32-bit) numbers. See `src/uu/mknod/src/mknod.rs:30-33` for the correct implementation

maybe move this function into uucore fs ?

sylvestre replied:
yeah

sylvestre commented on src/uucore/src/lib/features/fs.rs:
```diff
@@ -172,6 +172,30 @@ impl FileInformation {
+
+impl seal::Sealed for libc::dev_t {}
+impl DevExt for libc::dev_t {}
+
```
isn't enough ?

```suggestion
pub trait DevExt: Sealed + Copy + Into<libc::dev_t> {
    #[inline]
    fn major(self) -> libc::c_uint {
        libc::major(self.into())
    }

    #[inline]
    fn minor(self) -> libc::c_uint {
        libc::minor(self.into())
    }
}
```

sylvestre replied:
too easy :)

Alonely0 commented:
> please update mknod.rs also

There is also a makedev libc function, so it seems we can just use that.

Alonely0 commented:
It appears that macOS CI has been down for a while lol. There are multiple PRs (including this one) which have stalled jobs for anywhere in between 4 and 15 hours because of this. Is CI self-hosted (and fixable) or are we hostage to GitHub? [GitHub Status](https://www.githubstatus.com/) reports it's all fine... For what it's worth, last time I pushed, macOS ran successfully, and this last push was about Redox support (with conditional compilation for it), so this should be fine to merge.

Alonely0 commented:
It appears we ran into a filesystem race condition with atime and stime on the last runner of all the 11-hour-long CI lol (cries inside)... To avoid this in the future, I've just removed time checks and added a comment; it was naive of us to expect consistent filesystem timing anyway.

Alonely0 commented:
@sylvestre do you think this is ready to be merged or would any like any other changes? Lmk if I can help in any way.

Alonely0 commented:
I mean, it's done as an integration test in `test_stat.rs`. If you mean adding a unit test to the module that essentially does the same, it's trivially done.

Alonely0 commented:
> Thanks!

To you! I'm sorry I had unnecessarily made the PR harder to review on accident... Anyway, thanks for merging this!
